### PR TITLE
Add Rocky Linux

### PR DESCRIPTION
**Pull request by poorndm**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/mixlib/install/options.rb [hunk 1]
```diff
@@ -35,6 +35,7 @@ class InvalidOptions < ArgumentError; end
         powerpc
         ppc64
         ppc64le
+        rockylinux
```
Comment (vkarve-chef) [inline lib/mixlib/install/options.rb:38]: we can remove this entry, as we are only adding a platform and no new architecture.

File: lib/mixlib/install/generator/bourne/scripts/platform_detection.sh [hunk 2]
```diff
@@ -46,8 +46,13 @@ elif test -f "/etc/Eos-release"; then
 elif test -f "/etc/redhat-release"; then
   platform=`sed 's/^\(.\+\) release.*/\1/' /etc/redhat-release | tr '[A-Z]' '[a-z]'`
   platform_version=`sed 's/^.\+ release \([.0-9]\+\).*/\1/' /etc/redhat-release`
+  
+  if test "$platform" = "rocky linux"; then
+  	source /etc/os-release
+ 	os="${ROCKY_SUPPORT_PRODUCT}"
```
Comment (vkarve-chef) [inline lib/mixlib/install/generator/bourne/scripts/platform_detection.sh:52]: can we instead use `"${REDHAT_SUPPORT_PRODUCT}"` here as `"${ROCKY_SUPPORT_PRODUCT}"` will include version too, which we are separately fetching.

**Review by vkarve-chef — APPROVED** [review]

File: lib/mixlib/install/generator/bourne/scripts/platform_detection.sh [hunk 3]
```diff
@@ -46,8 +46,13 @@ elif test -f "/etc/Eos-release"; then
...
+  	platform_version="${ROCKY_SUPPORT_PRODUCT_VERSION}"
```
Comment (vkarve-chef) [inline lib/mixlib/install/generator/bourne/scripts/platform_detection.sh:54]: we can update platform to match the entry in [omnitruck](https://omnitruck.chef.io/platforms) by adding `platform="${ID}"`

Comment (poorndm) [reply lib/mixlib/install/generator/bourne/scripts/platform_detection.sh:54]: Its already in place. 
https://github.com/chef/mixlib-install/blob/main/lib/mixlib/install/generator/bourne/scripts/platform_detection.sh#L115

Comment (vkarve-chef) [reply lib/mixlib/install/generator/bourne/scripts/platform_detection.sh:54]: since that is in an `elif` block, it won't be applied in this block.

**Review by sean-simmons-progress — APPROVED** [review]

**Review by tpowell-progress — APPROVED** [review]

**Comment by tpowell-progress** [comment]
@sean-simmons-progress @vkarve-chef @poorndm Looks like we're good to merge this?

**Comment by sean-simmons-progress** [comment]
> @sean-simmons-progress @vkarve-chef @poorndm Looks like we're good to merge this?

merge away!

**Comment by mboutet** [comment]
@vkarve-chef @poorndm @tpowell-progress @sean-simmons-progress 

Is it possible that this change broke the bootstrap process for Rocky Linux? We're currently blocked from bootstrapping any new Rocky Linux nodes:

```
ERROR: -----> Installing Chef Omnibus (current/17.10.3)
     downloading https://omnitruck.chef.io/chef/install.sh
       to file /tmp/install.sh.8564/install.sh
     trying wget...
     rocky 8 x86_64
     Getting information for chef current 17.10.3 for rocky...
     downloading https://omnitruck.chef.io/current/chef/metadata?v=17.10.3&p=rocky&pv=8&m=x86_64
       to file /tmp/install.sh.8569/metadata.txt
     trying wget...
     ERROR 404
     Omnitruck artifact does not exist for version 17.10.3 on platform rocky
     
     Either this means:
        - We do not support rocky
        - We do not have an artifact for 17.10.3
     
     This is often the latter case due to running a prerelease or RC version of Chef
     or a gem version which was only pushed to rubygems and not omnitruck.
     
     You may be able to set your knife[:bootstrap_version] to the most recent stable
     release of Chef to fix this problem (or the most recent stable major version number).
     
     In order to test the version parameter, adventurous users may take the Metadata URL
     below and modify the '&v=<number>' parameter until you successfully get a URL that
     does not 404 (e.g. via curl or wget).  You should be able to use '&v=11' or '&v=12'
     successfully.
     
     If you cannot fix this problem by setting the bootstrap_version, it probably means
     that rocky is not supported.
     
     Metadata URL: https://omnitruck.chef.io/current/chef/metadata?v=17.10.3&p=rocky&pv=8&m=x86_64
     Starting the first Chef Infra Client Client run...
      [sudo] password for administrator: /tmp/chef_CRfctB/bootstrap.sh: line 297: chef-client: command not found
```

We were able to bootstrap Rocky Linux hosts just fine prior to this change.

**Comment by poorndm** [comment]
We haven't officially made a public announcement regarding the support for the rockyLinux platform. its still in works.

**Comment by mboutet** [comment]
@poorndm our Chef support assured us last year that it was supported and we've been using it without any issues until this PR broke the bootstrap process.

Why would you merge this PR before adding support in the omnitruck endpoint? Please revert this PR until support is added to the omnitruck endpoint or promptly add support to the endpoint.

**Comment by GeorgeWestwater** [comment]
@mboutet - Rocky linux is not an officially released; as such the mix-lib and omibus package are still in flux.

As Rocky Linux is designed to be binary-compatible with CentOS. This means Rocky Linux can run the same software as CentOS and be used for the same purposes. As, Rocky Linux is not a direct fork of CentOS, and we can not guarantee compatibility until we release a dedicated version. However, to be practical we know they have not drifted at the moment; thus a solution would be to use the CentoOS build of Chef to run on your Rocky Linux machine. 

However; I do not want state this is not yet released, and as such not on the officially supported list.

**Comment by anandhapadmanabhan** [comment]
Meanwhile you can download the binary directly via the omnitruck api and install the el build. https://omnitruck.chef.io/stable/chef/metadata?v=17&p=el&pv=8&m=x86_64

or edit the /etc/os-release file for installing and revert

`sed -e 's/^ID="[^"]*"/ID="redhat"/' -i /etc/os-release`

**Pull request merged**